### DISCUSSION TITLE
Question about using multiple @page decorators on the same function

### DISCUSSION THREAD

**Discussion by daya0576** [zauberzeug/nicegui#2889] [Q&A]
### Question

Code for Reproduction:
```python
@ui.page('/')
@ui.page('/gui')
def index_page():
    ui.label('Hello, world!')
```

Problem
- ✅ page can be accessed through `/` or `/gui`
- ❌ `Client.page_routes.values()` returns `['/']` instead of `['/', '/gui']`

Is this a bug or am I using it incorrectly? Thx!

**Comment by falkoschindler** [comment]
Hi @daya0576,

I guess this is a limitation of our `page_routes` dictionary. Its purpose is to find the correct route when writing `ui.navigate.to(index_page)`, which should still work.

What are you using `page_routes` for? If you need a mapping from callable to _all_ corresponding routes, we might need to find another representation. Otherwise I don't think there's an urgent need to change anything.

**Comment by daya0576** [comment]
Hi @falkoschindler 

> Its purpose is to find the correct route when writing ui.navigate.to(index_page), which should still work.

Got it, I use it to check whether the current path is valid or not, for example in `AuthMiddleware`:
```python
def init_gui_routes(fastapi_app: FastAPI):
    @app.middleware("http")
    async def AuthMiddleware(request: Request, call_next):
        client_page_routes = Client.page_routes.values()
        if not await user_check_token(app.storage.user.get("auth_token", None)):
            if (
                request.url.path in client_page_routes
                and request.url.path not in UNRESTRICTED_PAGE_ROUTES
            ):
                root_path = request.scope["root_path"]
                app.storage.user["referrer_path"] = request.url.path.removeprefix(
                    root_path
                )
                return RedirectResponse(request.url_for(login_page.__name__))
```

Could you provide a better solution in this case? Thanks!

**Reply by falkoschindler** [reply]
I see. You can get the route paths from the FastAPI `app`:

```py
from nicegui import app

client_page_routes = [route.path for route in app.routes if hasattr(route, 'path')]
```

Note that I'm using `hasattr` because - according to mypy - not every route has a path. Maybe there's an even better approach?

**Reply by daya0576** [reply]
Thx @falkoschindler, it works. I use `isinstance` for static checks, but not sure if there is any other type of Route.
```python
client_page_routes = [route.path for route in app.routes if isinstance(route, APIRoute)]
```
